iati/core/tests/test_data.py

```diff
@@ -406,3 +408,112 @@ def test_dataset_xml_str_source_around_line_invalid_context_lines(self, invalid_
         for line_num in range(0, num_lines_xml):
             with pytest.raises(TypeError):
                 data.source_around_line(line_num, invalid_value)
+
+
+class TestDatasetVersionDetection(object):
```

Comment:
Suggest adding a test to make sure you can't assign to the `version` property.

Reply:
It may be desired in the future once there are Dataset-creation features, though at present it's not really desired.

Reply:
Added in 1620a0a.